vaadin-grid-flow-integration-tests/src/main/java/com/vaadin/flow/component/grid/it/dataview/AbstractUndefinedSizeGridPage.java

```diff
@@ -0,0 +1,302 @@
+
+        MenuItem showLogs = menuItem.getSubMenu().addItem(
+                "Show fetch query logs",
+                event -> logPanel.setVisible(event.getSource().isChecked()));
+        showLogs.setCheckable(true);
+        showLogs.setChecked(true);
+    }
+
+    private void initEstimateOptions() {
+        MenuItem menuItem = menuBar.addItem("Size Estimate Configuration");
+        menuItem.setId(ESTIMATE_CALLBACK_MENU_ITEM_ID);
+        menuItem.getSubMenu().addItem(
+                "setUndefinedSize(SizeEstimateCallback) -> undefined size",
+                event -> switchToSizeEstimateCallback())
+                .setId(SIZE_ESTIMATE_CALLBACK_BUTTON_ID);
+
+        initialEstimateInput = new IntegerField(
+                "setUndefinedSize(int initialEstimate) -> undefined size:",
+                event -> grid.getLazyDataView()
+                        .withUndefinedSize(event.getValue()));
+        initialEstimateInput.setId(INITIAL_ESTIMATE_INPUT_ID);
+        initialEstimateInput.setWidthFull();
+
+        estimateInput = new IntegerField("Next Size Estimate:",
+                event -> sizeCallbackEstimate = event.getValue());
+        estimateInput.setId(SIZE_ESTIMATE_CALLBACK_INPUT_ID);
+        estimateInput.setPlaceholder("Test Default is +20% estimate");
+        estimateInput.setWidthFull();
+        menuItem.getSubMenu().add(initialEstimateInput, estimateInput);
+    }
+
+    private void initDataCommunicatorOptions() {
+        IntegerField pageSizeInput = new IntegerField("Page Size", event -> {
+            grid.setPageSize(event.getValue());
+        });
+        pageSizeInput.setValue(grid.getPageSize());
+        pageSizeInput.setWidthFull();
+
+        IntegerField bufferPagesInput = new IntegerField(
+                "# pages to increase size", event -> grid.getDataCommunicator()
+                        .setSizeIncreasePageCount(event.getValue()));
+        bufferPagesInput.setValue(
+                grid.getDataCommunicator().getSizeIncreasePageCount());
+        bufferPagesInput.setWidthFull();
+
+        MenuItem menuItem = menuBar.addItem("DataCommunicator Configuration");
+        menuItem.setId(DATA_COMMUNICATOR_MENU_ITEM_ID);
+        menuItem.getSubMenu().add(pageSizeInput, bufferPagesInput);
+    }
+
+    private int getSizeEstimate(SizeEstimateQuery<String, Void> query) {
+        Div log = new Div();
+        log.setId("log-" + sizeEstimateQueryCount);
+        log.setText(sizeEstimateQueryCount++ + ": Size "
+                + query.getPreviousSizeEstimate() + ", "
+                + query.getRequestedRangeEnd());
+        logPanel.addComponentAsFirst(log);
+        if (sizeCallbackEstimate < 1) {
+            // using +20% as the default
+            int previousSizeEstimate = query.getPreviousSizeEstimate();
+            if (previousSizeEstimate == 0) {
+                return DEFAULT_INITIAL_SIZE_ESTIMATE;
+            }
+            return new BigDecimal(previousSizeEstimate)
```

Comment:
We discussed it briefly before, but anyway I would ask why you use BigDecimal here? Don't we enough with a precision provided by `double`? It's a just an estimation which shouldn't be as much precise as possible. But, if we assume that we could use both `double` and `BigDecimal` here, and it's just an example for developers and the precision choice is up to them, then it's fine.

Reply:
This is a test class, what difference does it make ? At least it is consistent